internal/receiver/smartagentreceiver/config.go

```diff
@@ -16,18 +16,114 @@ package smartagentreceiver
 
 import (
 	"fmt"
+	"reflect"
+	"strings"
 
+	"github.com/signalfx/defaults"
+	_ "github.com/signalfx/signalfx-agent/pkg/core" // required to invoke monitor registration via init() calls
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/core/config/validation"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/spf13/viper"
 	"go.opentelemetry.io/collector/config/configmodels"
+	"gopkg.in/yaml.v2"
 )
 
 type Config struct {
 	configmodels.ReceiverSettings `mapstructure:",squash"`
+	monitorConfig                 config.MonitorCustomConfig
 }
 
-func (cfg *Config) validate() error {
-	if cfg.NameVal == "" || cfg.TypeVal == "" {
-		// validation placeholder for coverage
-		return fmt.Errorf("name and type are required")
+func (rCfg *Config) validate() error {
+	if rCfg.monitorConfig == nil {
+		return fmt.Errorf("you must supply a valid Smart Agent Monitor config")
 	}
+
+	monitorConfigCore := rCfg.monitorConfig.MonitorConfigCore()
+	if monitorConfigCore.IntervalSeconds == 0 {
+		monitorConfigCore.IntervalSeconds = 10
```

Comment:
Nit: we typically define config defaults as consts somewhere close to the top of the file for better visibility.